PKG Add galpy

jobovy opened the pull request:
<!-- Thank you for contributing to Pyodide! All improvements are welcome,
     so don't be afraid to make a PR. -->

### Description

Add [`galpy`](https://github.com/jobovy/galpy), a widely-used package in astrophysics for Galactic dynamics. `galpy` has been used [more than 450 times](https://docs.galpy.org/en/latest/#papers-using-galpy) in the astrophysical literature and is commonly used in undergraduate and graduate courses. It's also used in [my online textbook on galaxies](https://galaxiesbook.org/), for which I would eventually like to build a `pyodide`-backed interactive version (currently provided by Binder). So it would be great to add this to `pyodide`!

There are a few minor installation patches required that I will upstream for the next `galpy` release.

Thanks @[benjeffery](https://github.com/benjeffery) in #2548 for adding the GSL which made this easy!

### Checklists

<!-- Note:
     If you think some of these steps are not necessary for your PR,
     remove those checkboxes, or mark them as checked. If you keep unchecked checkboxes,
     we will assume that your PR is not ready to be merged  -->

- [ ] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry (will add once I know the PR number)
- [x] Add / update tests
- [x] Add new / update outdated documentation

hoodmane commented on packages/galpy/meta.yaml:
```diff
@@ -0,0 +1,36 @@
+    - patches/0002-Don-t-use-OpenMP-for-pyodide-build.patch
+build:
+  cflags: |
+    -I$(PYODIDE_ROOT)/packages/libgsl/build/libgsl-2.7
+  ldflags: |
+    -L$(PYODIDE_ROOT)/packages/libgsl/build/libgsl-2.7/.libs
+    -L$(PYODIDE_ROOT)/packages/libgsl/build/libgsl-2.7/cblas/.libs
```
Please use `pkg-config` for this. (See packages/msprime/meta.yaml.)

jobovy replied:
Ah, will do! I was looking at the `msprime` PR, didn't notice that there had been updates...

hoodmane commented:
Also, please add some tests.

jobovy commented:
Okay, I fixed the `pkg-config` use and added some basic tests. I had to exclude `galpy` from the node tests and I had to skip the check of the number of 'hiwire' keys. Both of these appear necessary because `galpy` loads `matplotlib` and it seems these checks don't work with `matplotlib` (the node one because there is no `matplotlib` backend for node, the hiwire one I don't understand, but it's skipped in `test_matplotlib.py`, so I just copied that code).

hoodmane commented:
The refcount check issue is that matplotlib adds a bunch of global JavaScript objects that are never going to be removed. The fix would be to load and import micropip in a fixture which would count the statically allocated objects as part of the baseline. But we've been too lazy to fix it.

hoodmane commented on packages/galpy/meta.yaml:
```diff
@@ -27,10 +27,6 @@ requirements:
 test:
   imports:
     - galpy
-    - galpy.potential
```
Put these back and add galpy to the list of `UNSUPPORTED_PACKAGES` for `node` on line 25 of `packages/test_packages_common.py`

jobovy commented:
Thanks @hoodmane and @rth! Yes, I'll apply these patches upstream so they shouldn't be necessary with the next release.
